### ISSUE TITLE
Throw error if missing something required.

### ISSUE THREAD

**Issue by ghost** [wankdanker/node-sftpjs#8]
macOS 
node 8.13

```js
const Client = require('sftpjs')
const connection = Client()
connection.on('ready', console.log)
connection.on('error', console.error)
connection.connect({username: ' '}) // not empty string but a space.
```

So in this case I'm not getting any error. In my config there was `host` instead of `hostname`
and there where no error. It took some time to figure out what is happening.

I think correct behavior will be to throw error if something required is missing.

**Comment by wankdanker** [comment]
I agree that the correct behavior should be to throw an error if something is missing. Unfortunately, this module just blindly passes along all connect options to ssh2.Client's connect function. So, either ssh2.Client doesn't have this behavior or we're somehow missing the throw.
